supporter-product-data/src/main/scala/com/gu/lambdas/UpdateDynamoLambda.scala

```diff
@@ -0,0 +1,113 @@
+package com.gu.lambdas
+
+import com.amazonaws.services.lambda.runtime.Context
+import com.gu.lambdas.UpdateDynamoLambda.{batchSize, writeToDynamo}
+import com.gu.model.Stage
+import com.gu.model.dynamo.SupporterRatePlanItem
+import com.gu.model.states.UpdateDynamoState
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger.Sanitizer
+import com.gu.services.{DynamoDBService, S3Service}
+import kantan.csv._
+import kantan.csv.ops._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, Future}
+
+trait TimeOutCheck {
+  def timeRemainingMillis: Int
+}
+
+class ContextTimeOutCheck(context: Context) extends TimeOutCheck {
+  override def timeRemainingMillis = context.getRemainingTimeInMillis
+}
+
+class UpdateDynamoLambda extends Handler[UpdateDynamoState, UpdateDynamoState] {
+  override protected def handlerFuture(input: UpdateDynamoState, context: Context) = {
+    SafeLogger.info(s"Starting write to dynamo task for ${input.recordCount} records from ${input.filename}")
+    writeToDynamo(Stage.fromEnvironment, input, new ContextTimeOutCheck(context))
+  }
+}
```

Comment:
you can use single abstact method syntax to reduce the boilerplate while keeping the nice naming, if you think it's still clear enough
```suggestion
trait TimeOutCheck {
  def timeRemainingMillis(): Int
}

class UpdateDynamoLambda extends Handler[UpdateDynamoState, UpdateDynamoState] {
  override protected def handlerFuture(input: UpdateDynamoState, context: Context) = {
    SafeLogger.info(s"Starting write to dynamo task for ${input.recordCount} records from ${input.filename}")
    writeToDynamo(Stage.fromEnvironment, input, () => context.getRemainingTimeInMillis)
  }
}
```